apps/src/componentLibrary/common/hooks/useDocumentKeydown.ts

```diff
@@ -0,0 +1,24 @@
+import {useEffect} from 'react';
+
+/**
+ * useDocumentKeydown
+ * A custom hook that attaches a `keydown` event listener to the document
+ * and calls the provided callback function when a key is pressed.
+ *
+ * @param callback - The function to call when a key is pressed.
+ */
+const useDocumentKeydown = (callback?: (event: KeyboardEvent) => void) => {
```

Comment:
Any reason the `callback` param can't be required? Is there a use case of using this hook without providing a callback?

Reply:
Yeah, that's intended for cases like this here: https://github.com/code-dot-org/code-dot-org/pull/62335/files#diff-72b1c4217c35caa27918d096d2223cf94255f34df4d76770c6b664f7801fc1a3R78 , where callback is optional, but it's not possible to put effect hook in if statement unfortunately